File: WebApiSpark/src/main/java/Model/Person.java
package Model;

import Util.DatabaseValues;

public class Person extends IncidentElement{

    public Person () {
        this (
                "",
                "",
                "",
                ""
        );
    }

    public Person (
            String firstName,
            String lastName,
            String personId,
            String phoneNumber
    ) {
        super (
                DatabaseValues.Table.PERSON,
                new DatabaseValues.Column []
                        {
                                DatabaseValues.Column.FIRST_NAME,
                                DatabaseValues.Column.LAST_NAME,
                                DatabaseValues.Column.PERSON_ID,
                                DatabaseValues.Column.PHONE_NUMBER
                        }
        );


        updateAttributeValue (
                DatabaseValues.Column.FIRST_NAME,
                firstName
        );
        updateAttributeValue (
                DatabaseValues.Column.LAST_NAME,
                lastName
        );
        updateAttributeValue (
                DatabaseValues.Column.PERSON_ID,
                personId
        );
        updateAttributeValue (
                DatabaseValues.Column.PHONE_NUMBER,
                phoneNumber
        );

    }

    @Override
    public String toSearchString() {
        return this.getAttributeValue( DatabaseValues.Column.FIRST_NAME ) + " " +
                this.getAttributeValue( DatabaseValues.Column.LAST_NAME ) + " " +
                this.getAttributeValue( DatabaseValues.Column.PHONE_NUMBER );
    }

    public DatabaseValues.Column getIDColumn () {
        return DatabaseValues.Column.PERSON_ID ;
    }

}

File: IRSClientAngular/src/app/component/person/person.ts
import { Config } from '../../util/config.service';
import { IncidentElement } from '../report/incident-element';

export class Person implements IncidentElement {
    table: string;
    attributes: PersonAttributes;

    constructor() {
        this.attributes = new PersonAttributes();
        this.table = Config.PersonTable;
    }

    toSearchString(): string {
        return this.attributes.FIRST_NAME 
        + " " + this.attributes.LAST_NAME 
        + " " + this.attributes.PHONE_NUMBER
    };
}

export class PersonAttributes {
    PERSON_ID: number;
    FIRST_NAME: string;
    LAST_NAME: string;
    PHONE_NUMBER:string;
}